FIX - PDF minor version number checking.

carlwilson opened the pull request:
- added test for invalid minor version number;
- added MessageConstant for invalid minor version;
- bumped PDF Module release details for release candidate;
- refactored `PdfHeader` class;
- JavaDoc for `PdfHeader`;
- unit tests and test resources for `PdfHeader`;
- added JHOVE RC PDF Module details to baseline; and
- improved result logging for Travis tests.
Closes #207

carlwilson committed:
Separate PDF Header parsing from Module.

carlwilson committed:
Dedicated PDF Header parsing class.

carlwilson committed:
PdfModule now uses dedicated Header parser.

carlwilson committed:
Merge remote-tracking branch 'origin/integration' into fix/pdf-header

carlwilson committed:
FIX - PDF minor version number checking.
- added test for invalid minor version number;
- added MessageConstant for invalid minor version;
- bumped PDF Module release details for release candidate;
- refactored `PdfHeader` class;
- JavaDoc for `PdfHeader`;
- unit tests and test resources for `PdfHeader`;
- added JHOVE RC PDF Module details to baseline; and
- improved result logging for Travis tests.
Closes #207

carlwilson committed:
FIX - Unnecessary if statement.

carlwilson committed:
Merge branch 'integration' into fix/pdf-header

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java:
```diff
@@ -1065,83 +1063,20 @@ protected final void initParse()
-            _pdfACompliant = false;
-        }
-        return true;
+		if (!header.isVersionValid()) {
```
Mixed indentation white-space.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/PdfHeader.java:
```diff
@@ -0,0 +1,184 @@
+/**
+ * 
+ */
```
Empty Javadoc.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/PdfHeader.java:
```diff
@@ -0,0 +1,184 @@
+		try {
+			minorVersion = getMinorVersion(this.versionString);
+		} catch (NumberFormatException nfe) {
+			// TODO : Do nothing for now, the version number is still invalid.
```
TODO: Nothing?

carlwilson replied:
The aim is to not break existing behaviour. This ensures that this case is handled as before as the minor version number is set too large and false is returned. I've left the TODO as I'm intending to polish this a little more, but it needs a new error messages and is outside of the PR scope.

asciim0 replied:
I wonder if in the long term view it is as easy as thinking about the minor version not being set too large, especially with PDF 2.0 now available. Couldn't it be instead checking the versionString against a list of currently available versions? So, as of today: 1.0, 1.1, 1.2, 1.3, ,1.4, 1.5., 1.6, 1.7, 2.0 ? 
Downside is that the list of available versions would have to be updated everytime a new version becomes available - but if the code instead states that the max correct minor version is 7, which is true for PDF 1.x, it would have to still be updated regarding the limitation of minor version for 2.x as there currently only is 2.0 (and this would again have to be updated once 2.1 becomes available).
Does that make any sense?

david-russo commented on jhove-modules/src/test/java/edu/harvard/hul/ois/jhove/module/pdf/PdfHeaderTest.java:
```diff
@@ -0,0 +1,101 @@
+/**
+ * 
+ */
```
Empty Javadoc.

carlwilson committed:
FIX - Review Comments
- replaced inconistent tabs with spaces; and
- removed empty JavaDoc headers.
Thanks to @david-russo for review.
Fixed the header JavaDoc generation on my IDE now also.

carlwilson committed:
Merge branch 'integration' into fix/pdf-header

carlwilson committed:
Merge branch 'integration' into fix/pdf-header
